components/board/genericlinux/board.go

```diff
@@ -254,11 +254,43 @@ func (b *sysfsBoard) AnalogReaderByName(name string) (board.AnalogReader, bool)
 }
 
 func (b *sysfsBoard) DigitalInterruptByName(name string) (board.DigitalInterrupt, bool) {
-	// TODO(RSDK-2345): If the name is numerical and doesn't already exist, create it here anyway.
+	if b.usePeriphGpio {
+		return nil, false // Digital interrupts aren't supported.
+	}
+
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
 	interrupt, ok := b.interrupts[name]
+	if ok {
+		return interrupt.interrupt, true
+	}
+
+	// Otherwise, the name is not something we recognize yet. If it appears to be a GPIO pin, we'll
+	// remove its GPIO capabilities and turn it into a digital interrupt.
+	gpio, ok := b.gpios[name]
 	if !ok {
 		return nil, false
 	}
+	if err := gpio.Close(); err != nil {
+		b.logger.Errorf("Unable to close GPIO pin to use as interrupt: %s", err)
+		return nil, false
+	}
+
+	defaultInterruptConfig := board.DigitalInterruptConfig{
+		Name: name,
+		Pin:  name,
+		Type: "basic",
```

Comment:
[nit] if type will be more than one type in the future, good to get ahead and use a `iota` or a 
` type interruptType string` definition:
```
    type CustomType string

    const (
        Foobar CustomType = "somestring"
    )
```

Reply:
The only valid types are "basic" and "servo," and Gautham thinks (and I agree, but his opinion is more informed than mine) that this was a mistake and there should only be basic ones, and "servo" digital interrupts were a mistake. I can't think of a time when we'd ever want a non-basic digital interrupt on a genericlinux board.

but I see your larger point, and I'll make it a named constant.